src/inbox/mod_inbox_rdbms_async.erl

```diff
@@ -0,0 +1,276 @@
+-module(mod_inbox_rdbms_async).
+
+-include("mod_inbox.hrl").
+
+-behaviour(mod_inbox_backend).
+
+-define(PER_MESSAGE_FLUSH_TIME, [?MODULE, per_message_flush_time]).
+-define(FLUSH_TIME, [?MODULE, flush_time]).
+
+-type async_task() ::
+    {set_inbox, mod_inbox:entry_key(), content(), pos_integer(), id(), integer()} |
+    {set_inbox_incr_unread, mod_inbox:entry_key(), content(), id(), integer(), Incrs :: pos_integer()} |
+    {remove_inbox_row, mod_inbox:entry_key()} |
+    {reset_unread, mod_inbox:entry_key(), id()}.
+
+%% API
+-export([init/2,
+         set_inbox/6,
+         set_inbox_incr_unread/5,
+         reset_unread/3,
+         remove_inbox_row/2,
+         remove_domain/2,
+         clear_inbox/3,
+         get_inbox/4,
+         get_inbox_unread/2,
+         get_entry_properties/2,
+         set_entry_properties/3]).
+
+%% Async callback
+-export([flush_inbox/2]).
+
+init(HostType, Opts0) ->
+    mod_inbox_rdbms:init(HostType, Opts0),
+    Opts1 = extend_opts(Opts0),
+    start_pool(HostType, Opts1),
+    ensure_metrics(HostType),
+    ok.
+
+extend_opts(Opts) ->
+    Merge = maps:merge(defaults(), maps:from_list(Opts)),
+    Opts1 = maps:to_list(Merge),
+    Opts2 = gen_mod:set_opt(flush_extra, Opts1, Merge),
+    add_callback(Opts2).
+
+add_callback(Opts) ->
+    gen_mod:set_opt(flush_callback, Opts, fun ?MODULE:flush_inbox/2).
+
+defaults() ->
+    #{flush_interval => 300,
+      batch_size => 100,
+      pool_size => 4 * erlang:system_info(schedulers_online)}.
```

Comment:
Would it be possible to move these defaults to the config spec?

Reply:
Will merge them into the config spec. When I was doing this code, the config spec defaults weren't merged to master yet 😄